openhgnn/trainerflow/DiffMG_trainer.py

```diff
@@ -0,0 +1,419 @@
+        adj_offset[ic_npy[:, 0] + offsets['i'], ic_npy[:, 1] + offsets['c']] = 1
+        adjs_offset['3'] = sp.coo_matrix(adj_offset)
+
+        ## iv
+        iv_npy = iv.to_numpy()[:, :2]
+        adj_offset = np.zeros((node_types.shape[0], node_types.shape[0]), dtype=np.float32)
+        adj_offset[iv_npy[:, 0] + offsets['i'], iv_npy[:, 1] + offsets['v']] = 1
+        adjs_offset['4'] = sp.coo_matrix(adj_offset)
+        pbar.update(1)
+
+        # f2 = open("./preprocessed/Amazon/adjs_offset.pkl", "wb")
+        # pickle.dump(adjs_offset, f2)
+        # f2.close()
+        return node_types, neg_ratings, ui_pos_train, ui_pos_val, ui_pos_test, adjs_offset, neg_train, neg_val, neg_test
+
+    def gen_neg(self,pos_pairs_offset_train, pos_pairs_offset_val, pos_pairs_offset_test, unconnected_pairs_offset,
+                neg_ratings_offset, neg_seed):
+
+        # pos_pairs_offset = np.load(os.path.join(prefix, "pos_pairs_offset.npz"))
+        # unconnected_pairs_offset = np.load(os.path.join(prefix, "unconnected_pairs_offset.npy"))
+        # neg_ratings_offset = np.load(os.path.join(prefix, "neg_ratings_offset.npy"))
+        np.random.seed(neg_seed)
+        train_len = pos_pairs_offset_train.shape[0]
+        val_len = pos_pairs_offset_val.shape[0]
+        test_len = pos_pairs_offset_test.shape[0]
+        pos_len = train_len + val_len + test_len
+
+        if pos_len > neg_ratings_offset.shape[0]:
+            indices = np.arange(unconnected_pairs_offset.shape[0])
+            assert (indices.shape[0] > pos_len)
+            np.random.shuffle(indices)
+            makeup = indices[:pos_len - neg_ratings_offset.shape[0]]
+            neg_ratings_offset = np.concatenate((neg_ratings_offset, unconnected_pairs_offset[makeup]), axis=0)
+            assert (pos_len == neg_ratings_offset.shape[0])
+        indices = np.arange(neg_ratings_offset.shape[0])
+        np.random.shuffle(indices)
+        train = neg_ratings_offset[indices[:train_len]],
+        val = neg_ratings_offset[indices[train_len:train_len + val_len]],
+        test = neg_ratings_offset[indices[train_len + val_len:pos_len]]
+        return train, val, test
+        # np.savez(os.path.join(prefix, "neg_pairs_offset"), train=neg_ratings_offset[indices[:train_len]],
+        #                                                     val=neg_ratings_offset[indices[train_len:train_len + val_len]],
+        #                                                     test=neg_ratings_offset[indices[train_len + val_len:pos_len]])
+
+        # ! Yelp 2
+        # ! Amazon 4
+        # ! Douban_Movie 6
+
+    def process(self):
+        if self.args.gpu > -1:
+            print("use gpu---------------")
+            torch.cuda.set_device(self.args.gpu)
+            torch.cuda.manual_seed(self.args.Amazon_train_seed)
+        else:
+            print("use cpu_______________")
+        np.random.seed(self.args.seed)
+        torch.manual_seed(self.args.seed)
```

Comment:
The seeds have already been initialized in the previous module.  I think we can remove these code.